### ISSUE TITLE
Ember 2.13.0 breaks nested keys in unit tests

### ISSUE THREAD

**Issue by Baukereg** [offirgolan/ember-cp-validations#502]
I try to upgrade a project from Ember 2.11.0 to Ember 2.13.0.

### Environment

- Ember Version: 2.13.0
- Ember CLI Version: 2.13.0
- Ember CP Validations Version: tested with both 3.2.1 and 3.3.1

### Steps to Reproduce

Say I have an object with a nested key validation:

```
import Ember from 'ember';
import { validator, buildValidations } from 'ember-cp-validations';

const Validations = buildValidations({
  'notification.subject': validator('presence', true)
});

export default Model.extend(Validations, {
  notification: null
});
```

This results in the next error when running a unit test:

```
Error: Assertion Failed: calling set on destroyed object: <(subclass of (unknown mixin)):ember311>._model = [object Null]
```

If I change 'notification.subject' to just 'notification', all works fine, so this problem seems to narrow down to just nested keys.

**Comment by offirgolan** [comment]
@Baukereg any chance you can recreate this in a twiddle?

**Comment by Baukereg** [comment]
I tried, but twiddle doesn't support 2.13.0 yet. So I created a small Ember project to illustrate the problem:

https://github.com/Baukereg/validation-nested-key

It only contains a _report_ model with validation of a nested key. The unit test will result in the error.

**Comment by Baukereg** [comment]
Apparently it has something to do with changes to Ember Data. I accidentally upgraded Ember Data from 2.11.0 to 2.13.0 in another project and the same error occurred. Hopefully I'll find some time tomorrow to dive deeper into this problem.

**Comment by marcemira** [comment]
Having the same issue using Ember Data 2.12.2, @Baukereg did you managed to find a solution? Thanks

**Comment by offirgolan** [comment]
Fixed in `v3.3.2`. @Baukereg thanks for bringing this to my attention 😄
